### PR TITLE
REC-32: Basic Nav Structure 

### PR DESCRIPTION
**Pull request by Musical-Bonesie**
## Resolves/Added: 

- Basic User Navigation between different pages/routes

### Testing:

1. Go to: http://localhost:3000
2. In terminal run: *pnpm dev*
3. Click on any links on the NavBar (some just route back to "/" until that component is made)
4. Click on the user icon in NavBar. A Drop-Down menu should appear, and when you click on _Profile_ it should take you to the users profile. 
5. When an individual recipe card is clicked it should route to the "/recipe" showing a single recipe page. 

#### Screen Shots of what it should look like: 

**Home Page:** 
image

**List of Recipes:** 
image

**User Profile **
image

### CODE REVIEW

**Comment by ArshSB** [comment]
I tested it and it works as advertised! Amazing work, navbar is looking great! I had one question, would having 'create account' as a dropdown selection make sense when the user is already logged in?

**Review by James-Firth — CHANGES_REQUESTED** [review]
Looking good!

❗ Watch out for scope creep going forward though. For example your recipe list page is looking fantastic, but that's adding extra work to this PR and then Ana and Matt will have a harder time tracking how long it takes people to do a task "like basic navigation". Or someone else could have been working on that task and then someone's time is lost

It happens by accident when one gets focused! We'll keep it (normally what happens when this happens) just good to keep in mind!

For a task like this that setup up foundational stuff and _could_ block someone (didn't!) I think it's best to stick with a terribly ugly, but simple text like the Profile Page without even blocking stuff out. That lets us get the barebones stuff moving forward and everyone has a file to work in (reduces conflicts) and then doing more layout or placeholder pages can come as a task after the other devs have a "canvas to work on" with the blank pages


🖼️ Can't comment on binary files so: Good spot to put those! I like having an assets folder like that. Not a concern with a test project like this but do you know if they're free to use? (or under public domain or creative commons license)

File: webapp/src/components/molecules/AdvancedCard/AdvancedCard.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { Card } from 'primereact/card';
+import { Button } from 'primereact/button';
+
+export interface routePath {
+  path?: {
+    pathName: string;
+  };
+}
+
+export default function AdvancedCard(props: routePath) {
```
[the remark this answers is not in the document]
Comment (Musical-Bonesie) [reply webapp/src/components/molecules/AdvancedCard/AdvancedCard.tsx:13]: Oh, I like the name!!

File: webapp/src/pages/HomePage/HomePage.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.homepage {
+  color: #f1faee;
+  background-image: url('../../assets//images/dish-back-ground-image.jpg');
```
[the remark this answers is not in the document]
Comment (Musical-Bonesie) [reply webapp/src/pages/HomePage/HomePage.css:3]: oof! I'll remove the background image.

File: webapp/src/components/molecules/UserDropDown/UserDropDown.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    },
+  ];
+
+  return (
+    <div className="card">
+      <TieredMenu model={items} popup ref={menu} id="overlay_tmenu" />
+      <Button
+        icon="pi pi-user"
+        // !Looked into using null with TS and couldn't figure out a way to resolve this issue with menu,current/useRef(null)
+        onClick={(event) => menu.current.toggle(event)}
```
[the remark this answers is not in the document]
Comment (Musical-Bonesie) [reply webapp/src/components/molecules/UserDropDown/UserDropDown.tsx:59]: Hmm TS seems moody when I try to use either option but I did get this to work: 
`const menu = useRef<any>(null);`

Think this is ok?

File: webapp/src/components/organisms/NavBar/NavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        <h2>Create Recipe</h2>
+      </Link>
+      <div className="nav-bar__elements">
+        <input
+          className="nav-bar__search-bar"
+          type="search"
+          placeholder="Search"
+        ></input>
+      </div>{' '}
+      {/* TODO For now this prop.user.name is hard coded until Auth/login is set up :-) */}
```
[the remark this answers is not in the document]
Comment (Musical-Bonesie) [reply webapp/src/components/organisms/NavBar/NavBar.tsx:29]: Would this fall into : 
_REC-29 Integrate Auth0 with the React app_ ?

File: webapp/src/components/molecules/AdvancedCard/AdvancedCard.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const navigate = useNavigate();
+
+  const navigateTo = (path: string) => {
+    console.log('I am a path', path);
```
[the remark this answers is not in the document]
Comment (Musical-Bonesie) [reply webapp/src/components/molecules/AdvancedCard/AdvancedCard.tsx]: :see_no_evil:

File: webapp/src/components/molecules/UserDropDown/UserDropDown.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+export default function UserDropDown(props: User) {
+  const navigate = useNavigate();
+
+  const navigateTo = (path: string) => {
+    navigate(path);
+  };
+
+  console.log(navigate, 'inside UserDropDown');
```
[the remark this answers is not in the document]
Comment (Musical-Bonesie) [reply webapp/src/components/molecules/UserDropDown/UserDropDown.tsx]: :see_no_evil:

File: webapp/src/components/molecules/AdvancedCard/index.tsx [hunk 7]
```diff
@@ -1 +0,0 @@
-export { default } from './AdvancedCard';
```
Comment (James-Firth) [inline webapp/src/components/molecules/AdvancedCard/index.tsx]: protip use `git mv $oldname $newname` so git understands it was a rename! Not required or anything just makes it a tiny bit easier to review

File: webapp/src/components/organisms/NavBar/NavBar.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (James-Firth) [reply webapp/src/components/organisms/NavBar/NavBar.tsx:29]: Yeah that sounds good

File: webapp/src/pages/HomePage/HomePage.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (James-Firth) [reply webapp/src/pages/HomePage/HomePage.css:3]: I just meant the double-slash in the path
```suggestion
  background-image: url('../../assets/images/dish-back-ground-image.jpg');
```

**Review by James-Firth — CHANGES_REQUESTED** [review]

File: webapp/src/components/molecules/AdvancedCard/index.tsx (hunk 7 above)
Comment (Musical-Bonesie) [reply webapp/src/components/molecules/AdvancedCard/index.tsx]: Oh! Never thought of that. I'll keep that in mind next time 🙂

**Comment by ArshSB** [comment]
I tested it out again and it's working well. I just have these suggestions:

- Sign out button in dropdown should direct to a route like '/logout' to let the user know they were successfully logged out
- Create account shows up when user is logged in. You might have placed it there as placeholder since auth0 wasn't set up. But I think it should only show up when the user is not logged in

You can use state placeholder to represent user logged in/logged out like I did for REC-30 and have the create account button show up based on that condition.
Overall I like the design and icons, it looks quite polished!

**Comment by James-Firth** [comment]
I think Arsh's comments make sense especially the user name. We can worry about the redirects during auth0 integration if it doesn't make sense to do now.

File: webapp/src/components/organisms/NavBar/NavBar.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <div className="nav-bar">
+      <Link to="/" className="nav-bar__logo">
+        <h2 className="nav-bar__text">GR RECIPES</h2>
+      </Link>
+      <Link to="/recipe-list" className="nav-bar__link">
+        <h2>Home</h2>
+      </Link>
+      <Link to="/home" className="nav-bar__link">
+        <h2>Recipes</h2>
```
Comment (James-Firth) [inline webapp/src/components/organisms/NavBar/NavBar.tsx:16]: I think Home and Recipes got flip-flopped either text or link location

**Review by James-Firth — APPROVED** [review]
shipit :shipit:

**Pull request merged**